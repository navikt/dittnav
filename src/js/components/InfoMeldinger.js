import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F } from 'react-intl';

import PaabegynteSoknader, { PaabegynteSoknaderType } from 'js/components/meldinger/PaabegynteSoknader';
import RegStatusLink from 'js/components/RegStatusLink';
import Meldekort, { MeldekortType } from 'js/components/Meldekort';
import EtterregistreringMeldekort from 'js/components/EtterregistreringMeldekort';
import AgMeldinger from 'js/components/meldinger/AgMeldinger';
import NavMeldinger from 'js/components/meldinger/NavMeldinger';
import MinInnboks, { MinInnboksType } from 'js/components/meldinger/MinInnboks';
import InformasjonsMeldinger, { InfoMessagesType } from 'js/components/meldinger/InformasjonsMeldinger';

class InfoMeldinger extends Component {
  render() {
    return (
      <section className="infomeldinger-list">
        <h1 className="vekk"><F id="dittnav.infomeldinger.varsler" /></h1>
        <Meldekort meldekort={this.props.meldekort} />
        <EtterregistreringMeldekort ettereg={this.props.meldekort}/>
        <RegStatusLink isRegisteredAtIArbeid={this.props.isRegisteredAtIArbeid} />
        {!this.props.isInactive ? <AgMeldinger agMessagesCount={this.props.agMessagesCount} /> : null}
        <NavMeldinger navMessagesCount={this.props.navMessagesCount} />
        <InformasjonsMeldinger infoMessages={this.props.infoMessages} isMeldeKortUser={this.props.isMeldeKortUser} />
        <PaabegynteSoknader paabegynteSoknader={this.props.paabegynteSoknader} />
        <MinInnboks mininnboks={this.props.mininnboks} />
      </section>
    );
  }
}

InfoMeldinger.propTypes = {
  paabegynteSoknader: PaabegynteSoknaderType,
  isRegisteredAtIArbeid: PropTypes.bool,
  meldekort: MeldekortType,
  navMessagesCount: PropTypes.number,
  agMessagesCount: PropTypes.number,
  infoMessages: InfoMessagesType,
  isMeldeKortUser: PropTypes.bool,
  isInactive: PropTypes.bool.isRequired,
  mininnboks: MinInnboksType,
};

InfoMeldinger.defaultProps = {
  paabegynteSoknader: null,
  isRegisteredAtIArbeid: null,
  meldekort: null,
  navMessagesCount: 0,
  agMessagesCount: 0,
  infoMessages: null,
  isMeldeKortUser: false,
  mininnboks: [],
};

export default InfoMeldinger;
