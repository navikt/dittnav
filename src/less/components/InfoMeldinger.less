@import (reference) '~nav-frontend-lenker-style/src/lenker-mixins';

.infomeldinger-list {
  border-radius: 0.25rem;
  margin-top: 2.5rem;
  margin-right: 4.375rem;
  margin-left: 4.375rem;
  .blokk-xl();

  .infomelding {
    margin-bottom: 1rem;
    padding: 0.75rem;

    .lenkepanel__innhold {
      padding: 0;
    }
  }
}

.varslinger-inngang-wrapper {
  text-align: center;
}

.varslinger-inngang {
  display: inline-block;

  &__chevron {
    color: @navBla;
    display: inline-block;
    margin-left: 0.15em;
  }

  &:hover {
    text-decoration: underline;
    .varslinger-inngang__chevron {
      transform: translateX(3px) rotate(-90deg);
    }
  }
}

#varslinger-inngang__lenke-id {
  border-bottom: 0;
  .lenke-mixin();

  &:not(:hover) {
    text-decoration: none;
  }
}

@media screen and (max-width: 700px) {
  .infomeldinger-list {
    margin-right: 0;
    margin-left: 0;
  }
}

@media screen and (max-width: @mobile-breakpoint) {
  .varslinger-inngang {
    font-size: 1rem;
  }
}
