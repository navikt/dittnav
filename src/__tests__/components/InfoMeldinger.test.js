import * as React from 'react';
import InfoMeldinger from 'js/components/InfoMeldinger';
import wrapIntl from 'js/IntlTestHelper';
import ShallowRenderer from 'react-test-renderer/shallow';

test('render empty InfoMeldinger component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<InfoMeldinger />));
  const component = renderer.getRenderOutput();
  expect(component).toMatchSnapshot();
});
