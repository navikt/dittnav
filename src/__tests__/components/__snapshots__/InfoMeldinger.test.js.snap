// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty InfoMeldinger component 1`] = `
<InfoMeldinger
  beskjeder={null}
  innbokser={null}
  innlogging={null}
  meldekort={null}
  mininnboks={Array []}
  oppgaver={null}
  paabegynteSoknader={null}
/>
`;
